src/execution/mod.rs

```diff
@@ -441,33 +691,26 @@ fn worker_main<A: Allocate>(
 /// >>> run_main(flow, ManualConfig(input_builder), output_builder)  # doctest: +ELLIPSIS
 /// (...)
 ///
-/// See `bytewax.run()` for a convenience method to not need to worry
-/// about input or output builders.
-///
 /// See `bytewax.spawn_cluster()` for starting a cluster on this
 /// machine with full control over inputs and outputs.
 ///
 /// Args:
 ///
 ///     flow: Dataflow to run.
 ///
-///     input_config: Input config of type Manual or Kafka. See `bytewax.inputs`.
-///
-///     output_builder: Returns a callback function for each worker
-///         thread, called with `(epoch, item)` whenever and item
-///         passes by a capture operator on this process.
+///     epoch_config: A custom epoch config. You probably don't need
```

Comment:
This still feels funny to me, but i know we still have a lot of moving parts and maybe . . . maybe it just has to be funny, at least for now, but maybe eventually we'll be able to extract some of the testing mechanisms.

This is a 💭 only

Reply:
Yeah. I'm not sure what to do here because this is a subtle part of the API. Another idea might be to have this take the `ClockConfig` instead? Then you could use the `TestingClockConfig` to programmatically set the time in tests. But I don't think it could be exactly the same clock as the windowing because the idea of watermarks doesn't apply here. So then we'd be adding in a special new clock which seems actually more confusing than an `EpochConfig`. Perhaps we can streamline this later, but I do think the input-epoch separation is a step forward.

Reply:
Agree agree